all-benchmarks/control/CFReT_benchmarks/CFReT-repo_benchmarks.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Benchmark Data Compilation Notebook
+#
+# In this Notebook, our objective is to compile benchmark data from the `\bulk_benchmarks` and `\single_benchmarks` directories, generating two distinct `.csv` files with raw data.
+# These files will serve as datasets for direct comparison with the benchmarks of the CytoSnake workflows.
+#
+
+# ## Imports
+
+# In[1]:
+
+
+import json
+import pathlib
+import sys
+from datetime import datetime
+
+import pandas as pd
+
+sys.path.append("../../../")
+from src.benchmark_utils import get_benchmark_files  # noqa
+
+# # Parameters Used in this Notebook
+#
+# Follow parameters used in this notebook.
+
+# In[2]:
+
+
+# inputs
+working_dir = pathlib.Path().resolve()
+benchmark_dir = pathlib.Path("./benchmarks").resolve(strict=True)
+
+# outputs paths
+# single_benchmark_csv = working_dir / "single_benchmarks.csv"
+
+
+# ## Loading all JSON files
+#
+# Here we are loading all the JSON files.
+# The file name structure of the JSON files is `{Plate_name}_{type}_{process}_benchmarks.json`.
+# Also we are loading the file size information.
+
+# In[3]:
+
+
+single_json_files = list(get_benchmark_files(benchmark_dir, ext="json"))
+
+# loading json file that contains file size information
+with open("./file_size.json", encoding="utf-8", mode="r") as content:
+    plate_size = json.load(content)
+
+
+# In[4]:
+
+
+# applying time format
+tformat = "%Y-%m-%d %H:%M:%S.%f"
+
+# collecting all data
+raw_benchmark_data = []
+
+# iterating each json file and extract data
+for single_json_file in single_json_files:
+    # collecting data from just file name
+    plate_name = single_json_file.stem.split("_CFReT_")[0]
+    file_size = plate_size[plate_name]
+    process_name = single_json_file.stem.split("_CFReT_")[1].split("_benchmark")[0]
```

Comment:
The naming of the benchmarks files will have to be consistent, since most of the identification of the data types and processes are structured within the file names.